source/fundamentals/crud/read-operations/count.txt

```diff
@@ -0,0 +1,256 @@
+
+The following ``Student`` class is a Plain Old CLR/Class Object, or
+:ref:`poco <csharp-poco>`,  that models the documents in this
+collection.
+
+.. literalinclude:: /includes/code-examples/count-documents/Student.cs
+   :start-after: start-grades-struct
+   :end-before: end-grades-struct
+   :language: csharp
+   :dedent:
+
+.. note::
+
+   The documents in the ``grades`` collection use the camel-case naming
+   convention. The examples in this guide use a ``ConventionPack``
+   to deserialize the fields in the collection into Pascal case and map them to
+   the properties in the ``Student`` class. 
+
+   To learn more about custom serialization, see
+   :ref:`csharp-custom-serialization`. 
+   
+.. _csharp-accurate-count:
+
+Accurate Count
+--------------
+
+To count the number of documents that match your :ref:`query filter <csharp-specify-query>`, use the
+``CountDocuments()`` method. If you pass an empty query filter, this method
+returns the total number of documents in the collection.
+
+Example 
+~~~~~~~
+
+The following example counts the number of documents where the
+``finalGrade`` is less than ``80``:
+
+.. io-code-block::
+   :copyable: true
+
+   .. input::
+      :language: csharp
+
+      var filter = Builders<BsonDocument>.Filter.Lt("finalGrade", 80);
+      var count = coll.CountDocuments(filter);
+      
+      Console.WriteLine("Number of documents with a final grade less
+      than 80: " + count);
+
+   .. output::
+      :language: none
+      :visible: false
+
+      Number of documents with a final grade less than 80: 2
+
+Modify Behavior
+~~~~~~~~~~~~~~~
+
+You can modify the behavior of ``CountDocuments()`` by passing in a
+``CountOptions`` type. If you don't specify any options, the driver uses
+its default values.
+
+The ``CountOptions`` type allows you to configure options with the
+following properties:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Property
+     - Description
+
+   * - ``Collation``
+     - | The type of language collation to use when sorting results.  
+       | Default: ``nil``
+
+   * - ``Hint`` 
+     - | The index to use to scan for documents to count. 
+       | Default: ``nil``
+
+   * - ``Limit`` 
+     - | The maximum number of documents to count. 
+       | Default: ``0`` 
+
+   * - ``MaxTime``
+     - | The maximum amount of time that the query can run on the server.
+       | Default: ``nil``
+
+   * - ``Skip`` 
+     - | The number of documents to skip before counting.
+       | Default: ``0``
+
+.. tip::
+
+   When you use ``CountDocuments()`` to return the total number of documents in a
+   collection, MongoDB performs a collection scan. You can avoid a collection scan and
+   improve the performance of this method by using a hint to take advantage of the built-in index on
+   the ``_id`` field. Use this technique only when calling ``CountDocuments()``
+   with an empty query parameter.
+
+   .. code-block:: csharp
+      :emphasize-lines: 1, 3
```

Comment:
Q: why is the filter highlighted?

Reply:
To emphasize the use of an Empty filter, but I can remove if it's distracting from the point.